core-processor/src/executor.rs

```diff
@@ -258,13 +112,12 @@ fn prepare_memory<A: ProcessorExt, M: Memory>(
     // Set initial data for pages
     for (page, data) in pages_data.iter_mut() {
         mem.write(page.offset(), data)
-            .map_err(|err| ExecutionErrorReason::InitialDataWriteFailed(*page, err))?;
+            .unwrap_or_else(|err| unreachable!("Cannot write initial data for {page:?}: {err}"));
```

Comment:
Is it really `unreachable`?

Reply:
Here and so on.

Reply:
Yes. It means we corrupted contract memory somehow